diff --git a/WandB.py b/WandB.py
deleted file mode 100644
index e69de29..0000000
diff --git a/disvae/training.py b/disvae/training.py
index 2566c71..58df81d 100644
--- a/disvae/training.py
+++ b/disvae/training.py
@@ -86,7 +86,6 @@ class Trainer():
             self.logger.info('Epoch: {} Average loss per image: {:.2f}'.format(epoch + 1,
                                                                                mean_epoch_loss))
             
-            wandb.log({"epoch": (epoch+1), "mean_epoch_loss": mean_epoch_loss})
             self.losses_logger.log(epoch, storer)
 
             if self.gif_visualizer is not None:
@@ -133,7 +132,6 @@ class Trainer():
 
                 t.set_postfix(loss=iter_loss)
                 t.update()
-        wandb.log({"epoch": (epoch+1), "iter_loss": iter_loss})
         mean_epoch_loss = epoch_loss / len(data_loader)
         return mean_epoch_loss
 
@@ -163,7 +161,7 @@ class Trainer():
         except ValueError:
             # for losses that use multiple optimizers (e.g. Factor)
             loss = self.loss_f.call_optimize(data, self.model, self.optimizer, storer)
-
+        
         return loss.item()
 
 
@@ -191,6 +189,8 @@ class LossesLogger(object):
         for k, v in losses_storer.items():
             log_string = ",".join(str(item) for item in [epoch, k, mean(v)])
             self.logger.debug(log_string)
+            wandb.log({"epoch": epoch, "log_string": log_string})
+            print({"epoch": epoch, "log_string": log_string})
 
 
 # HELPERS
diff --git a/wandb/latest-run b/wandb/latest-run
index b2a0404..a620e98 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20201223_145420-2uuqzszp
\ No newline at end of file
+run-20201226_155206-1fk07tm3
\ No newline at end of file
